### PR TITLE
Implement `wait_for_complete()` for Speaker class

### PR DESCRIPTION
**Pull request by dvonthenen**
## Proposed changes

Addresses: https://github.com/deepgram/deepgram-python-sdk/issues/469

Please see issue for details...

## Types of changes

What types of changes does your code introduce to the community Python SDK?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

NA

## Release Notes

- **New Features**
	- Introduced `PLAYBACK_DELTA` constant for enhanced audio playback management.
	- Added `wait_for_complete` method to the `Speaker` and WebSocket client classes for improved control over audio playback completion.

- **Improvements**
	- Updated configuration options for speaker playback timing.
	- Enhanced the handling of asynchronous operations in text-to-speech examples for a more responsive experience.

### CODE REVIEW

File: deepgram/audio/speaker/constants.py [hunk 1]
```diff
@@ -4,9 +4,12 @@
...
 from ...utils import verboselogs
 
-# Constants for microphone
+# Constants for speaker
 LOGGING = verboselogs.WARNING
 TIMEOUT = 0.050
 CHANNELS = 1
 RATE = 16000
 CHUNK = 8194
+
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/audio/speaker/constants.py:13]: Noted.

File: deepgram/audio/speaker/speaker.py [hunk 2]
```diff
@@ -56,6 +64,7 @@ def __init__(
         rate: int = RATE,
         chunk: int = CHUNK,
         channels: int = CHANNELS,
+        last_play_delta_in_ms: int = PLAYBACK_DELTA,
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/audio/speaker/speaker.py:67]: We should keep all the required parameters together. The only reason why ` pull_callback` and `pull_callback` are marked optional is that they are actually required BUT in some cases, we actually need them to be set after initializing the class.

File: deepgram/clients/speak/v1/websocket/client.py [hunk 3]
```diff
@@ -96,6 +96,11 @@
             channels = self._config.options.get("speaker_playback_channels")
             if channels is None:
                 channels = CHANNELS
+            playback_delta_in_ms = self._config.options.get(
+                "speaker_playback_delta_in_ms"
+            )
+            if playback_delta_in_ms is None:
+                playback_delta_in_ms = PLAYBACK_DELTA
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/clients/speak/v1/websocket/client.py:103]: Fine as is.

File: deepgram/clients/speak/v1/websocket/async_client.py [hunk 4]
```diff
@@ -590,6 +597,21 @@ async def clear(self) -> bool:
...
+        """
+        This method will block until the speak is done playing sound.
+        """
+        self._logger.spam("AsyncSpeakWebSocketClient.wait_for_complete ENTER")
+
+        if self._speaker is None:
+            self._logger.error("speaker is None. Return immediately")
+            return
+
+        loop = asyncio.get_event_loop()
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/clients/speak/v1/websocket/async_client.py:610]: Keeping this for now.

File: deepgram/clients/speak/v1/websocket/async_client.py [hunk 5]
```diff
@@ -93,6 +93,11 @@
             channels = self._config.options.get("speaker_playback_channels")
             if channels is None:
                 channels = CHANNELS
+            playback_delta_in_ms = self._config.options.get(
+                "speaker_playback_delta_in_ms"
+            )
+            if playback_delta_in_ms is None:
+                playback_delta_in_ms = PLAYBACK_DELTA
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/clients/speak/v1/websocket/async_client.py:100]: Keeping this for now.

**Review by jpvajda — APPROVED** [review]

**Pull request merged**